infrastructure/src/main/java/org/corfudb/infrastructure/SequencerServer.java

```diff
@@ -98,6 +105,12 @@
      */
     private final Map<UUID, Long> streamTailToGlobalTailMap = new HashMap<>();
 
+    /**
+     * Per streams map and their corresponding address space (an address space is defined by the stream's addresses
+     *  and its latest trim mark)
+     */
```

Comment:
As I understand this map is a copy of the map used in LogMetadata? Can we get rid of one of them?

Reply:
One lives in the log unit, the other in the sequencer... one is used for recovery, the other for actual query.

Reply:
so, why we need 4 copies of the same map?

Reply:
I'm not sure why you say we have 4 copies. Stream maps are kept in the sequencer (which are the ones queried by clients when syncing the address space of a stream). And another is kept in the log unit used for sequencer recovery, in the event of a shutdown these are rebuilt on init and used to bootstrap the sequencer (in the same way we do with stream tails).